Add an explicit dependency on the `logger` gem

Earlopain opened the pull request:
This is getting the same treatment as `base64`, `mutex_m`, etc. In Ruby 3.4 it will start to warn: https://github.com/ruby/ruby/commit/d7e558e3c48c213d0e8bedca4fb547db55613f7c

See for example #48907 

I also removed two requires that seem unnessesary: The test doesn't access `::Logger` and the other one requires `as/logger` right after.

zzak commented on activesupport/CHANGELOG.md:
```diff
@@ -1,2 +1,5 @@
+Add `logger` as a dependency since it is a bundled gem candidate for Ruby 3.5
+
+*Earlopain*
```
Could you fix?

https://buildkite.com/rails/rails/builds/108043#018fe820-3592-43e8-9bad-25785f217dd9/1272

```
activesupport/CHANGELOG.md:1 CHANGELOG header must start with '*' and 3 spaces
Add `logger` as a dependency since it is a bundled gem candidate for Ruby 3.5
^^^^
activesupport/CHANGELOG.md:3 CHANGELOG line must be indented 4 spaces
*Earlopain*
^^^^
13 changelogs inspected, 2 offenses detected
```

Earlopain replied:
Yes, my bad. Not good about running this beforehand and you get no notification for non-gh action failures

zzak replied:
No worries, though not sure we need notifications, anyone else reviewing this PR should have noticed it.

I enjoy pointing these things out in hopes it unblocks others, CI is often a source of confusion ranging from "why is this flaky test failing" to just completely ignoring it.

Earlopain replied:
Nah, I just mean the buildin notifications from github itself. If a github action fails you get info via email and push notificaion on mobile. It's per user _and_ repo, so everyone can set it up how they like it, unfortunatly this doesn't happen for external services which is sad (but somewhat understandable).

I have the bad habit of pushing and letting CI (not) notify me of things, especially since the rails suite is so big

Earlopain committed:
Add an explicit dependency on the `logger` gem

This is getting the same treatment as `base64`, `mutex_m`, etc.
In Ruby 3.4 it will start to warn: https://github.com/ruby/ruby/commit/d7e558e3c48c213d0e8bedca4fb547db55613f7c

Remoce require from two files that don't seem to need it
